src/openms/include/OpenMS/FORMAT/OMSFile.h

```diff
@@ -0,0 +1,344 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/CONCEPT/ProgressLogger.h>
+#include <OpenMS/KERNEL/FeatureMap.h>
+#include <OpenMS/METADATA/ID/IdentificationData.h>
+
+#include <QtSql/QSqlDatabase>
+#include <QtSql/QSqlError>
+#include <QtSql/QSqlQuery>
+
+namespace OpenMS
+{
+  /**
+      @brief This class supports reading and writing of OMS files.
+
+      OMS files are SQLite databases consisting of several tables.
+  */
+  class OPENMS_DLLAPI OMSFile: public ProgressLogger
+  {
+  public:
+    /// Constructor (with option to set log type)
+    explicit OMSFile(LogType log_type = LogType::NONE):
+      log_type_(log_type)
+    {
+      setLogType(log_type); // @TODO: move logging to OMSFileLoad
+    }
+
+    /** @brief Write out an IdentificationData object to SQL-based OMS file
+     *
+     * @param filename The output file
+     * @param id_data The IdentificationData object
+     */
+    void store(const String& filename, const IdentificationData& id_data);
+
+    /** @brief Write out a feature map to SQL-based OMS file
+     *
+     * @param filename The output file
+     * @param features The feature map
+     */
+    void store(const String& filename, const FeatureMap& features);
+
+    /** @brief Read in a OMS file and construct an IdentificationData object
+     *
+     * @param filename The input file
+     * @param id_data The IdentificationData object
+     */
+    void load(const String& filename, IdentificationData& id_data);
+
+    /** @brief Read in a OMS file and construct a feature map
+     *
+     * @param filename The input file
+     * @param features The feature map
+     */
+    void load(const String& filename, FeatureMap& features);
+
+  protected:
+    using Key = qint64;
+
+    LogType log_type_;
+
+    // convenience functions:
+
+    static bool tableExists_(const String& db_name, const String& table_name);
+
+    static void raiseDBError_(const QSqlError& error, int line,
+                              const char* function, const String& context);
+
+    // store helper class:
+    class OMSFileStore: public ProgressLogger
```

Comment:
@hendrikweisser can you move the helper class out of the main interface into a separate header file using the `OpenMS::Internal` namespace?